### DISCUSSION TITLE
Add a lazyboundary/ lazy load margin to all Nextjs images?

### DISCUSSION THREAD

**Discussion by Github010101011** [vercel/next.js#37482] [Help]
I want to add a lazyBoundary of 40px on all images rather than individually adding it as a tag.  Is this possible? if so how can i configure a default lazyBoundary to all Nextjs images for lazyloading?

**Comment by icyJoseph** [comment]
Perhaps not the best solution, but you can make an Image component, which takes all props that next/image takes, and overrides the lazy boundary to the value you want. 

```tsx
import Image, {ImageProps} from "next/image";

export const CustomNextImage (props: ImageProps) => {
    return <Image {...props lazyBoundary="40px" />
}
```

You could also, with Typescript, limit the props


```tsx
import Image, {ImageProps} from "next/image";

export const CustomNextImage (props:  Omit<ImageProps, "lazyBoundary">) => {
    return <Image {...props lazyBoundary="40px" />
}
```

This is all in the spirit of not messing with the underlying components, and keeping things flexible as well as explicitly telling other devs that this image component has custom behavior.

There are a few other combinations you can try, but the idea remains, create a custom component that enforces the prop value you want. If you are worried about import statements looking bad because of relative import syntax, look into absolute imports with Typescript. It'll make it possible to `import {CustomImage} from "components/CustomImage";`
